# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/x11/xfree86/ppc-imps2-wheel.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---


Hey cool - patching this paricular line seems to become a full-time job ...
or a good candidate for a cron job ;-)

Otherwise negative movements (0xff, ...) will be interpreted as 255 (...)
and since it is above 7 assigned to 0 :-(((

  - Rene Rebe <rene@rocklinux.org>

--- xc/programs/Xserver/hw/xfree86/input/mouse/mouse.c.orig	2003-10-10 13:06:30.000000000 +0200
+++ xc/programs/Xserver/hw/xfree86/input/mouse/mouse.c	2003-12-06 16:41:17.000000000 +0100
@@ -1386,7 +1386,7 @@
 		      (pBuf[0] & 0x80) >> 3;        /* button 5 */
 	    dx = (pBuf[0] & 0x10) ?    pBuf[1]-256  :  pBuf[1];
 	    dy = (pBuf[0] & 0x20) ?  -(pBuf[2]-256) : -pBuf[2];
-	    dz = (char)(pBuf[3] | ((pBuf[3] & 0x08) ? 0xf8 : 0));
+	    dz = (signed char)(pBuf[3] | ((pBuf[3] & 0x08) ? 0xf8 : 0));
 	    if ((pBuf[3] & 0xf8) && ((pBuf[3] & 0xf8) != 0xf8)) {
 		if (pMse->autoProbe) {
 		    SetMouseProto(pMse, PROT_EXPPS2);
